tests/ResponseSerializerTest.php

```diff
@@ -46,4 +47,12 @@ public function it_can_serialize_and_unserialize_a_response()
 
         $this->assertEquals('testValue', $unserializedResponse->headers->get('testHeader'));
     }
+
+    /** @test */
+    public function it_throws_when_something_something_else_than_a_response_is_unserialized()
```

Comment:
Rename to `it_throws_an_exception_when_something_else_than_a_response_is_unserialized`